src/braket/pennylane_plugin/translation.py

```diff
@@ -340,6 +340,17 @@ def _(h: qml.Hermitian):
     return observables.Hermitian(h.matrix)
 
 
+@_translate_observable.register
+def _(p: qml.Projector):
```

Comment:
Can this be written as a tensor product of single-qubit projectors instead?

Reply:
@trbromley I’m kind of surprised this is needed here; in PennyLane core, the presence of `qml.expval(qml.Projector(i))` should automatically return `marginal_prob[i]` for any device that inherits from `QubitDevice`, you simply need to add `”Projector”` to the list of supported observables of the device.

Of course, if the device supports directly computing a probability of a single basis state, this would be a more efficient approach.

Reply:
Most of our backends do support computing the probability one way or the other - either through direct amplitudes for the shots=0 (analytic) case on simulators, or sampling probabilities for shots > 0 (TN1 is the only exception). Unfortunately the managed simulators don't support marginal probabilities for analytic mode right now, which is probably why `marginal_prob` doesn't work in general.

This does make me think that we should start differentiating our implementation based on the specific Braket backend. For example, the list of supported gates in the Braket plugin is independent of the backend, which is not great.

For now though, I suspect that returning a tensor product of single-qubit projectors is easiest.

Reply:
Good idea @licedric, I have adopted the tensor product approach now (b8cf0500d6adce29ecb529c4aa71838c7d9e1b67 and 862dd37211c94715cd7cc28faac767389c40694e). Also, it would be nice to leverage the existing probability support in Braket. My initial fix was to edit [this](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/f19daa528e29edc7cf5afa6b02932b6e575519e6/src/braket/pennylane_plugin/translation.py#L286) line:
```python
if return_type is ObservableReturnTypes.Probability or isinstance(observable, qml.Projector):
```
Doing so means that we return the full probability distribution, and then we can select the element of the distribution in the [statistics](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/f19daa528e29edc7cf5afa6b02932b6e575519e6/src/braket/pennylane_plugin/braket_device.py#L171) method. However, I couldn't get this to work for the edge case discussed in [this](https://github.com/PennyLaneAI/pennylane/pull/1356/files#r641298375) comment, so in the end thought it'd be simpler to use the `observables.Hermitian` approach for now.

@josh146 - I agree that ideally this would work using existing PL functionality. However, I'm not sure that `BraketQubitDevice` (which inherits from `QubitDevice`) ever ends up calling the `marginal_prob` method since it defines its own `execute` method. As far as I can tell, the projector machinery in https://github.com/PennyLaneAI/pennylane/pull/1356 only supports devices that use some of the builtin functionality like the `expval` method.

Reply:
Thanks, that makes sense @trbromley!

Reply:
Update: changing to use tensor products appears to have a problematic edge case. Check out the following code:
```python
import pennylane as qml

# dev = qml.device("default.qubit", wires=3)
dev = qml.device("braket.local.qubit", wires=3)

@qml.qnode(dev)
def f():
    qml.RX(0.1, wires=0)
    qml.RX(0.1, wires=1)
    return qml.expval(qml.PauliZ(0) @ qml.Projector([1, 0], wires=[1, 2]))

f()
```
This does not work with the latest changes, but did work on commit c5cd843d89608e3a2daebad4e140d19836bb0d43 and before :thinking: (it also works fine with `default.qubit`). The error is:
```
ValueError: Operator qubit count 1 must be equal to size of target qubit set QubitSet([Qubit(1), Qubit(2)])
```
@licedric any idea what the issue is? I believe what is going on behind the scenes is:
- We pass through the [translation](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/src/braket/pennylane_plugin/translation.py) logic
- First we see that `qml.PauliZ(0) @ qml.Projector([1, 0], wires=[1, 2])` is a tensor product, so we then translate its elements `qml.PauliZ(0)` and `qml.Projector([1, 0], wires=[1, 2])`.
- Translating `qml.Projector([1, 0], wires=[1, 2])` gives itself a tensor product over two qubits.
- We hence calculate `observables.TensorProduct([_translate_observable(factor) for factor in t.obs])` with one term in `t.obs` being `observables.Z()` and the other being itself a tensor product over 2 qubits.

Reply:
That seems to be a potential shortcoming / bug in Braket - tensor products are not expanded in Braket. @speller26 do you think this is something we can fix quickly, or should we go with the original Hermitian matrix approach here for the time being?

Reply:
I'd be happy to revert to the original approach for now to prioritize passing the test suite. That approach will only cause problems when projecting onto many qubits, so perhaps it wouldn't even be noticed as a shortcoming initially.

Reply:
Alternatively, in `translation.py`, change line 347 to

```python
    return reduce(lambda x, y: x @ y, [_translate_observable(factor) for factor in t.obs])
```

(of course importing `reduce` from `functools`). This will give a flat tensor product.

Reply:
Thanks @speller26! Have made the change (f9c0b1fcce8b4a115d2d67ea65196d98cd6916e9) and it works.